### ISSUE TITLE
[BUG] MUI Data Grid no longer horizontally scaling (also on official examples)

### ISSUE THREAD

**Issue by PercivalFigaro** [refinedev/refine#6077]
### Describe the bug

When the browser window size is reduced, then horizontal scaling does not properly occur (table keeps full width and flows under the sidebar). A hard page refresh correctly resizes the table. Increasing window size scales the Data Grid properly, decreasing window size does not. This bug started occurring last Friday, possibly connected to a Chrome update.

### Steps To Reproduce

1. Use latest version of Google Chrome
2. Open up any of Refine's official examples using MUI and its Data Grid
3. Reduce the browser window size by making Chrome dev tools larger
4. Instead of scaling, the columns of the Data Grid do not reduce in size and full table width is maintained

### Expected behavior

The Data Grid should horizontally scale, not flow under the sidebar

### Packages

    "@mui/icons-material": "^5.8.3",
    "@mui/lab": "^5.0.0-alpha.85",
    "@mui/material": "^5.8.6",
    "@mui/x-data-grid": "^6.6.0",
    "@refinedev/cli": "^2.16.33",
    "@refinedev/core": "^4.51.0",
    "@refinedev/devtools": "^1.2.3",
    "@refinedev/kbar": "^1.3.12",
    "@refinedev/mui": "^5.17.0",
    "@refinedev/react-hook-form": "^4.8.20",
    "@refinedev/react-router-v6": "^4.5.11",
    "@refinedev/simple-rest": "^5.0.8",

**Comment by alicanerdurmaz** [comment]
Hello @PercivalFigaro, I can confirm that there are no issues with Safari, but it is not working properly on Chrome. We will look into this, thank you.

Table example:
https://example.mui.admin.refine.dev/orders

**Comment by joe-noumena** [comment]
Think sticking in a width 100% is required on the themedLayoutV2 
![image](https://github.com/refinedev/refine/assets/156899063/e049ea22-ee19-41e5-961b-04cf41663c6b)

**Comment by joe-noumena** [comment]
Can also be fixed at the theme level

```
    MuiCssBaseline: {
      styleOverrides: {
        ".MuiBox-root": {
          width: "100%", // Set the width to 100% as can break responsiveness of the layout
        },
      },
    },
```

**Comment by PercivalFigaro** [comment]
Thanks for the info. This workaround makes it scale horizontally (tried on the theme level), but only after disappearing behind the sidebar. Also it breaks some other styling on the page, notice the logout button, which worked nicely before:

![chrome_alXOE3xw9z](https://github.com/refinedev/refine/assets/97585394/36894c14-d30f-4de0-9403-a1aa4206578f)

I'm sure that the data grid worked out of the box before - it stopped working around June 20nd or 21st without any dependency changes in my codebase.

**Comment by joe-noumena** [comment]
Its probably best to eject the ThemedLayoutV2 and make the update there as overriding the box at theme level could have some other consequences. Was more for quick example

**Comment by PercivalFigaro** [comment]
@alicanerdurmaz, as a side note, the problem is more apparent in the official `auth-material-ui` example rather than the foods app:

![chrome_kJOwLJxVJ8](https://github.com/refinedev/refine/assets/97585394/41eca51d-632b-4e64-a292-1efcdad7ac2d)

**Comment by rners01** [comment]
I'm not sure if it's a proper fix but disabling the `overflow: clip` specifically for tablet, desktop media fixes the problem.
Although it's the internal code style of `ThemedLayoutV2` so I can't simply rewrite it

<img width="511" alt="image" src="https://github.com/refinedev/refine/assets/115642795/d63e8924-b834-4ae1-9d76-414520b772f1">
<img width="523" alt="image" src="https://github.com/refinedev/refine/assets/115642795/4d69b64d-62db-4970-86de-48239baa4660">

**Comment by PercivalFigaro** [comment]
Hey, thanks for the comment input - however, it doesn't do the trick for me

**Comment by rners01** [comment]
Hey @PercivalFigaro,
Did you set the `overflow: auto` for the `Box`?
With this example 
https://example.mui.admin.refine.dev/orders?pageSize=10&current=1
<img width="1162" alt="image" src="https://github.com/refinedev/refine/assets/115642795/51880c45-3143-44d7-a702-c617e55e1179">

**Comment by joe-noumena** [comment]
Its width 100% that will fix it on chrome

**Comment by EvgenyKut** [comment]
Hello! I have same urgent issue. Do we have any result solution? 

I can fix it in devTools, but how fix in the code? 
![2024-07-12 18 38 01](https://github.com/user-attachments/assets/015431d0-7fe5-4aec-9673-47e35a214cc4)

**Comment by aliemir** [comment]
Hey everyone, sorry for the issue. I've opened a PR #6144 including a fix for with a small tweak in the content section of the layout. I also included a preview deploy for the `finefoods-material-ui` example in the PR, https://deploy-preview-6144--finefoods-admin-mui.netlify.app Can someone check and validate if this fixes the issue? 🙏 

Until this is released, temporarily swizzling the `ThemedLayoutV2` component and applying the fix can be used as a workaround 🚀
